src/lib.rs

```diff
@@ -29,6 +29,38 @@ impl fmt::Display for Error {
 
 impl std::error::Error for Error {}
 
+pub struct Settings {
+    dav1d_settings: Dav1dSettings,
+}
+
+impl Default for Settings {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl Settings {
+    pub fn new() -> Self {
+        unsafe {
+            let mut dav1d_settings = mem::MaybeUninit::uninit();
+
+            dav1d_default_settings(dav1d_settings.as_mut_ptr());
+
+            Self {
+                dav1d_settings: dav1d_settings.assume_init(),
+            }
+        }
+    }
+
+    pub fn set_max_threads(self: &mut Self, n_threads: u32) {
+        self.dav1d_settings.n_tile_threads = n_threads as i32;
+    }
+
+    pub fn set_max_frame_delay(self: &mut Self, max_frame_delay: u32) {
```

Comment:
You might also want to add
 - getters for these fields
 - setters for the other simple fields (i.e. not allocator and logger)